### PR TITLE
Mentioned that other data types can be used to index objects

### LINKED ISSUE #33872: Mention that other data types can be used to index objects
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures

### What specific section or headline is this issue about?

Objects -> Properties

### What information was incorrect, unhelpful, or incomplete?

The statement "Property keys are either strings or symbols." is incorrect because it omits the fact that numbers can also be used directly as property keys in JavaScript objects. While it is true that numbers are implicitly converted to strings, they can still be assigned as property keys. For clarity and completeness, this behavior should be explicitly mentioned in the documentation.

### What did you expect to see?

I expected the documentation to state that property keys in JavaScript can be strings, numbers, or symbols. Specifically, the documentation should clarify that while numbers used as property keys are internally converted to strings, they can still be used directly in the code. This clarification would provide a more accurate and complete understanding of how property keys work in JavaScript objects.

### Do you have any supporting links, references, or citations?

[MDN Web Docs - Objects](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Working_with_Objects)

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/javascript/data_structures`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/data_structures/index.md
* Last commit: https://github.com/mdn/content/commit/c3c81e79b3bc798ffb69e80e93c10f74fb27467f
* Document last modified: 2023-12-20T02:24:39.000Z

</details>

### PR DESCRIPTION
**Pull request by Ashish-CodeJourney**
### Description

Property keys in JavaScript objects can be strings, symbols, numbers, or even objects themselves, allowing for a wide range of data types to be used as keys for indexing objects.

This revised statement reflects the broader range of data types that can be used as property keys in JavaScript objects.

[Mentioned that other data types can be used to index objects](https://github.com/Ashish-CodeJourney/content/commit/4c3d5e3b0e039d378da579c2d05e8aff5241f13a)

### Additional details

https://github.com/mdn/content/issues/33872#issuecomment-2143800965

### Related issues and pull requests

Fixes #33872

### CODE REVIEW

File: files/en-us/web/javascript/data_structures/index.md [hunk 1]
```diff
@@ -155,7 +155,8 @@ In computer science, an object is a value in memory which is possibly referenced
 
 ### Properties
 
-In JavaScript, objects can be seen as a collection of properties. With the [object literal syntax](/en-US/docs/Web/JavaScript/Guide/Grammar_and_types#object_literals), a limited set of properties are initialized; then properties can be added and removed. Object properties are equivalent to key-value pairs. Property keys are either [strings](#string_type) or [symbols](#symbol_type). Property values can be values of any type, including other objects, which enables building complex data structures.
+In JavaScript, objects can be seen as a collection of properties. With the [object literal syntax](/en-US/docs/Web/JavaScript/Guide/Grammar_and_types#object_literals), a limited set of properties are initialized; then properties can be added and removed. Object properties are equivalent to key-value pairs. Property keys in JavaScript objects can be [strings](#string_type), [symbols](#symbol_type), [numbers](#number_type), or even [objects](#object_type) themselves, allowing for a wide range of data types to be used as keys for indexing objects.
+Property values can be values of any type, including other objects, which enables building complex data structures.
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/data_structures/index.md:159]:
```suggestion
In JavaScript, objects can be seen as a collection of properties. With the [object literal syntax](/en-US/docs/Web/JavaScript/Guide/Grammar_and_types#object_literals), a limited set of properties are initialized; then properties can be added and removed. Object properties are equivalent to key-value pairs. Property keys in JavaScript objects can be [strings](#string_type) or [symbols](#symbol_type). When other types (such as numbers) are used to index objects, the values are implicitly converted to strings. Property values can be values of any type, including other objects, which enables building complex data structures.
```

File: files/en-us/web/javascript/data_structures/index.md [hunk 2]
```diff
@@ -155,7 +155,8 @@ In computer science, an object is a value in memory which is possibly referenced
 
 ### Properties
 
-In JavaScript, objects can be seen as a collection of properties. With the [object literal syntax](/en-US/docs/Web/JavaScript/Guide/Grammar_and_types#object_literals), a limited set of properties are initialized; then properties can be added and removed. Object properties are equivalent to key-value pairs. Property keys are either [strings](#string_type) or [symbols](#symbol_type). Property values can be values of any type, including other objects, which enables building complex data structures.
+In JavaScript, objects can be seen as a collection of properties. With the [object literal syntax](/en-US/docs/Web/JavaScript/Guide/Grammar_and_types#object_literals), a limited set of properties are initialized; then properties can be added and removed. Object properties are equivalent to key-value pairs. Property keys in JavaScript objects can be [strings](#string_type), [symbols](#symbol_type), [numbers](#number_type), or even [objects](#object_type) themselves, allowing for a wide range of data types to be used as keys for indexing objects.
+ Property values can be values of any type, including other objects, which enables building complex data structures.
```
[the remark this answers is not in the document]
Comment (Ashish-CodeJourney) [reply files/en-us/web/javascript/data_structures/index.md:159]: @Josh-Cena does it looks good now ?

**Comment by Josh-Cena** [comment]
I specifically asked in my review comment that you should not say "property keys can be numbers" but that "you can use numbers to index objects and the number will be converted to a string". I don't know why your last commit reverted it to the previous state.

**Comment by Ashish-CodeJourney** [comment]
> I specifically asked in my review comment that you should not say "property keys can be numbers" but that "you can use numbers to index objects and the number will be converted to a string". I don't know why your last commit reverted it to the previous state.

I don't know either, why it reverted, but you can review latest commit

**Comment by Josh-Cena** [comment]
You can go to https://github.com/mdn/content/pull/33892/files to see what your whole PR is proposing to change. You should make it incorporate my suggestion

**Review by Ashish-CodeJourney — COMMENTED** [review]
@Josh-Cena incorporated

**Comment by Josh-Cena** [comment]
Have you checked the files tab? I do see any difference

**Comment by Ashish-CodeJourney** [comment]
> Have you checked the files tab? I do see any difference

Yes i checked it looks good (as you suggested earlier)

File: files/en-us/web/javascript/data_structures/index.md (hunk 1 above)
Comment (Josh-Cena) [inline files/en-us/web/javascript/data_structures/index.md:159]:
```suggestion
In JavaScript, objects can be seen as a collection of properties. With the [object literal syntax](/en-US/docs/Web/JavaScript/Guide/Grammar_and_types#object_literals), a limited set of properties are initialized; then properties can be added and removed. Object properties are equivalent to key-value pairs. Property keys in JavaScript objects can be [strings](#string_type) or [symbols](#symbol_type). When other types (such as numbers) are used to index objects, the values are implicitly converted to strings. Property values can be values of any type, including other objects, which enables building complex data structures.
```

**Review by Josh-Cena — CHANGES_REQUESTED** [review]
Here was what I suggested

**Comment by Ashish-CodeJourney** [comment]
> Here was what I suggested

Okay got it why it reverted to previous state 
does it looks good now ?

File: files/en-us/web/javascript/data_structures/index.md [hunk 3]
```diff
@@ -155,7 +155,7 @@ In computer science, an object is a value in memory which is possibly referenced
 
 ### Properties
 
-In JavaScript, objects can be seen as a collection of properties. With the [object literal syntax](/en-US/docs/Web/JavaScript/Guide/Grammar_and_types#object_literals), a limited set of properties are initialized; then properties can be added and removed. Object properties are equivalent to key-value pairs. Property keys are either [strings](#string_type) or [symbols](#symbol_type). Property values can be values of any type, including other objects, which enables building complex data structures.
+In JavaScript, objects can be seen as a collection of properties. With the [object literal syntax](/en-US/docs/Web/JavaScript/Guide/Grammar_and_types#object_literals), a limited set of properties are initialized; then properties can be added and removed. Object properties are equivalent to key-value pairs. Property keys in JavaScript objects can be [strings](#string_type) or [symbols](#symbol_type). When other types (such as numbers) are used to index objects, the values are implicitly converted to strings. Property values can be values of any type, including other objects, which enables building complex data structures.
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/data_structures/index.md:158]:
```suggestion
In JavaScript, objects can be seen as a collection of properties. With the [object literal syntax](/en-US/docs/Web/JavaScript/Guide/Grammar_and_types#object_literals), a limited set of properties are initialized; then properties can be added and removed. Object properties are equivalent to key-value pairs. Property keys are either [strings](#string_type) or [symbols](#symbol_type). When other types (such as numbers) are used to index objects, the values are implicitly converted to strings. Property values can be values of any type, including other objects, which enables building complex data structures.
```

**Review by Josh-Cena — APPROVED** [review]
Thanks

**Pull request merged**